File: Two Pointer (MS)/125_Valid_Palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  const newString = s.replace(/[^\w\s]|_/g, "").replace(/\s+/g, "");
  var leftPointer = 0;
  var rightPointer = 0;
  for (let i = 0; i < newString.length / 2; i++) {
    if (
      newString[i].toLowerCase() ===
      newString[newString.length - 1 - i].toLowerCase()
    ) {
      rightPointer += 1;
    }
    leftPointer += 1;
  }

  return rightPointer === leftPointer ? true : false;
};

console.log(isPalindrome("race a car"));

File: Two Pointer (MS)/125_Valid_Palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        filtered_word = ''.join(c for c in s if c.isalnum()).lower()
        length = len(filtered_word)

        left = 0
        right = length - 1
        while left < right:
            if filtered_word[left] != filtered_word[right]:
                return False

            left += 1
            right -= 1

        return True


if __name__ == "__main__":
    solution = Solution()
    print(solution.isPalindrome("race a car"))